q2_fmt/tests/test_engraftment.py

```diff
@@ -1121,10 +1122,76 @@ def test_rename_features_with_blank_label(self):
                                        reference_column="Ref",
                                        subject_column="subject")
         _rename_features(data=feature_peds_df, level_delimiter=";")
-        print(feature_peds_df)
         Fs1 = feature_peds_df.set_index("id").at['Feature 1 __',
                                                  'subject']
         Fs2 = feature_peds_df.set_index("id").at['Feature 2',
                                                  'subject']
         self.assertEqual("1", Fs1)
         self.assertEqual("2", Fs2)
+
+
+class TestBoot(TestBase):
+    def test_high_donor_overlap(self):
+        metadata_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3',
+                   'donor1', 'donor2', 'donor3'],
+            'Ref': ['donor1', 'donor2', 'donor3', float("Nan"), float("Nan"),
+                    float("Nan")],
+            'subject': ['sub1', 'sub2', 'sub3', float("Nan"), float("Nan"),
+                        float("Nan")],
+            'group': [1, 1, 1, float("Nan"), float("Nan"),
+                      float("Nan")],
+            "Location": [float("Nan"), float("Nan"),
+                         float("Nan"), 'test', 'test',
+                         'test']}).set_index('id')
+
+        table_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3',
+                   'donor1', 'donor2', 'donor3'],
+            'Feature1': [1, 0, 0, 1, 0, 0],
+            'Feature2': [0, 1, 0, 0, 1, 0],
+            'Feature3': [0, 0, 1, 0, 0, 1]}).set_index('id')
+        metadata = Metadata(metadata_df)
+
+        stats = peds_bootstrap(metadata=metadata,
+                               table=table_df,
+                               time_column="group",
+                               reference_column="Ref",
+                               subject_column="subject",
+                               bootstrap_replicates=999)
+        real_median = stats["A:measure"].values
+        fake_median = stats["B:measure"].values
+        self.assertGreater(real_median, fake_median)
```

Comment:
Do these tests pass all the time, or do you see intermittent failures? Ideally failures won't ever occur, but they should be _really_ rare if there are intermittent failures.

Reply:
I have not  experienced intermittent failures. Did you?